gpkit/nomials.py

```diff
@@ -45,14 +47,14 @@ def __init__(self, exps=None, cs=1, varlocsandkeys=None,
             cs = exps
             exps = {}
         if (isinstance(cs, Numbers)
-           and isinstance(exps, Strings + (VarKey, NoneType, dict))):
+           and (exps is None or isinstance(exps, Strings + (VarKey, dict)))):
             # building a Monomial
             if isinstance(exps, VarKey):
                 exp = {exps: 1}
                 units = exps.descr["units"] if "units" in exps.descr else None
             elif exps is None or isinstance(exps, Strings):
                 exp = {VarKey(exps, **descr): 1}
-                descr = exp.keys()[0].descr
+                descr = list(exp)[0].descr
```

Comment:
`exp.keys()` also generates a list dynamically. But, agreed. We don't have a great way to turn an exp into a varkey.